pkg/session/session.go

```diff
@@ -206,7 +206,14 @@ func GetOrCreate(ctx context.Context, params *Params) (*Session, error) {
 
 	// Assign the datacenter if one was specified.
 	if params.datacenter != "" {
-		dc, err := session.Finder.Datacenter(ctx, params.datacenter)
+		var dc *object.Datacenter
+		dcRef := object.ReferenceFromString(params.datacenter)
+		if dcRef != nil {
+			dc = object.NewDatacenter(client.Client, *dcRef)
+			_, err = dc.ObjectName(ctx)
```

Comment:
should it be verified if the returned obj name matches the expected one?

Reply:
Actually the idea here was less to check the DC name, but instead force a vsphere API call to happen to guarantee that the passed MOID ref for the datacenter is valid.

We cannot match the expected because in this case we don't know the expected DC name, (eg RicardoDC), but the MOID. What I can do anyway is maybe add an info log here saying the Ref was resolved to DC name returned by that ObjectName call, wdyt?

Reply:
+1 for adding the log

Reply:
+ 1. Debug level log would be helpful, whether it's using the string as MOID or real name.